extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java

```diff
@@ -1103,15 +1103,15 @@ private static void producePersistenceUnitDescriptorFromConfig(
             p.put(JAKARTA_SHARED_CACHE_MODE, SharedCacheMode.NONE);
         }
 
-        // Hibernate Validator integration: we force the callback mode to have bootstrap errors reported rather than validation ignored
-        // if there is any issue when bootstrapping Hibernate Validator.
-        if (capabilities.isPresent(Capability.HIBERNATE_VALIDATOR)) {
-            if (persistenceUnitConfig.validation().enabled()) {
-                descriptor.getProperties().setProperty(AvailableSettings.JAKARTA_VALIDATION_MODE,
-                        ValidationMode.CALLBACK.name());
-            } else {
-                descriptor.getProperties().setProperty(AvailableSettings.JAKARTA_VALIDATION_MODE, ValidationMode.NONE.name());
-            }
+        if (!persistenceUnitConfig.validation().enabled()) {
+            descriptor.getProperties().setProperty(AvailableSettings.JAKARTA_VALIDATION_MODE, ValidationMode.NONE.name());
+        } else {
+            descriptor.getProperties().setProperty(
+                    AvailableSettings.JAKARTA_VALIDATION_MODE,
+                    persistenceUnitConfig.validation().modes()
+                            .stream()
+                            .map(Enum::name)
+                            .collect(Collectors.joining(",")));
```

Comment:
btw.. do we want this to be included in the reactive processor? or reactive does not play nice with validation? (that was that other question that @yrodiere had: why does DDL work for reactive..)

Reply:
I have no idea, I haven't been involved in the Hibernate Reactive extension.

Either it's an oversight or it was not working. @DavideD would know better but, if we do implement it for HR also, I think it should be done in another PR anyway.

Reply:
I merged this one as that's something I want in 3.19. I let you discuss the situation in HR and see if it needs to be addressed in a follow-up PR.

Reply:
Created https://github.com/quarkusio/quarkus/issues/46327 for the Hibernate Reactive part.